doc/sfp-cmis/Interface-Link-bring-up-sequence.md

```diff
@@ -184,18 +188,294 @@ if transceiver is not present:
+
+Following infra will ensure port re-initialization by xcvrd in case of syncd/swss/orchagent crash
+Pre-requisites for the infra to work:
+  - APPL_DB should be cleared after syncd/swss/orchagent crash/restart, config reload and cold reboot
+  - APPL_DB should be retained after 
+    - pmon crash/restart
+    - xcvrd crash/restart (does not apply to xcvrd restart triggered due to a different docker container crash/restart)
+    - warm reboot
+
+1. XCVRD main thread init
+	- XCVRD main thread creates the key CMIS_REINIT_REQUIRED in PORT_TABLE:\<port\> (APPL_DB) with value as true for ports which do NOT have this key present 
+	- XCVRD main thread creates the key NPU_SI_SETTINGS_SYNC_STATUS in PORT_TABLE:\<port\> (APPL_DB) with value NPU_SI_SETTINGS_DEFAULT for ports which do NOT have this key present.  
+      - For transceivers which do not require NPU SI settings, NPU_SI_SETTINGS_SYNC_STATUS will stay with value NPU_SI_SETTINGS_DEFAULT  
+
+    Following table describes the various values for NPU_SI_SETTINGS_SYNC_STATUS  
+A transceiver is classified as CMIS SM driven transceiver if its module type is CMIS and it does not have flat memory  
+
+| Value                    | Modifier thread and event                                                                                                                                                                                           | Consumer thread and purpose                                                                     |
+| ------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------- |
+| NPU_SI_SETTINGS_DEFAULT  | 1\. XCVRD main thread during cold start of XCVRD<br>2. SfpStateUpdateTask during transceiver removal                                                                                                                | XCVRD main thread during boot-up for deciding to notify NPU SI settings                         |
+| NPU_SI_SETTINGS_NOTIFIED | 1\. SfpStateUpdateTask while updating and notifying the NPU SI settings for non-CMIS SM driven transceivers (this approach was chosen to preserve the existing behavior for non-CMIS SM driven transceivers)<br> 2. CmisManagerTask while updating and notifying the NPU SI settings for CMIS SM driven transceivers | Not being used currently                                                                        |
+| NPU_SI_SETTINGS_DONE     | Orchagent after applying the SI settings                                                                                                                                                                            | CmisManagerTask for proceeding from CMIS_STATE_NPU_SI_SETTINGS_WAIT to CMIS_STATE_DP_INIT state |
```

Comment:
@mihirpat1 does platform that does not need SI settings, do those port's CMIS state machine transition to DP init?

Reply:
Yes - For platforms or ports not requiring NPU SI settings, the CMIS state machine will transition to DP init state (i.e. skip transition to CMIS_STATE_NPU_SI_SETTINGS_WAIT)